src/coreReporters/runnerReporter.js

```diff
@@ -0,0 +1,69 @@
+var RunnerReporter = function (emitter) {
+  this.emitter = emitter;
+  this.testResult = [];
+  this.failedCount = 0;
+};
+
+RunnerReporter.prototype.register = function () {
+  jasmine.getEnv().addReporter(this);
+};
+
+RunnerReporter.prototype.jasmineStarted = function () {
+  // Need to initiate startTime here, in case reportSpecStarting is not
+  // called (e.g. when fit is used)
+  this.startTime = new Date();
+
+  browser.plugins_.addPlugin({
```

Comment:
The idea is to have a single flag, restartBrowserBetweenSpecs ( for consistent naming with config.specs and config.suites params) that will be handled in the TaskRunner. And if set, will restart the browser before taking the next task.